pkg/util/mount/mount_helper_common.go

```diff
@@ -31,52 +31,22 @@ import (
 func CleanupMountPoint(mountPath string, mounter Interface, extensiveMountPointCheck bool) error {
 	// mounter.ExistsPath cannot be used because for containerized kubelet, we need to check
 	// the path in the kubelet container, not on the host.
-	pathExists, pathErr := PathExists(mountPath)
+	pathExists, _ := PathExists(mountPath)
 	if !pathExists {
 		klog.Warningf("Warning: Unmount skipped because path does not exist: %v", mountPath)
 		return nil
 	}
-	corruptedMnt := IsCorruptedMnt(pathErr)
-	if pathErr != nil && !corruptedMnt {
-		return fmt.Errorf("Error checking path: %v", pathErr)
-	}
-	return doCleanupMountPoint(mountPath, mounter, extensiveMountPointCheck, corruptedMnt)
-}
-
-// doCleanupMountPoint unmounts the given path and
-// deletes the remaining directory if successful.
-// if extensiveMountPointCheck is true
-// IsNotMountPoint will be called instead of IsLikelyNotMountPoint.
-// IsNotMountPoint is more expensive but properly handles bind mounts within the same fs.
-// if corruptedMnt is true, it means that the mountPath is a corrupted mountpoint, and the mount point check
-// will be skipped
-func doCleanupMountPoint(mountPath string, mounter Interface, extensiveMountPointCheck bool, corruptedMnt bool) error {
-	if !corruptedMnt {
-		var notMnt bool
-		var err error
-		if extensiveMountPointCheck {
-			notMnt, err = IsNotMountPoint(mounter, mountPath)
-		} else {
-			notMnt, err = mounter.IsLikelyNotMountPoint(mountPath)
-		}
 
-		if err != nil {
-			return err
-		}
-
-		if notMnt {
-			klog.Warningf("Warning: %q is not a mountpoint, deleting", mountPath)
-			return os.Remove(mountPath)
-		}
-	}
-
-	// Unmount the mount path
-	klog.V(4).Infof("%q is a mountpoint, unmounting", mountPath)
 	if err := mounter.Unmount(mountPath); err != nil {
-		return err
+		klog.Warningf("Warning: Unmount mountPath %s failed with %v", mountPath, err)
+	}
+	var notMnt bool
+	var mntErr error
+	if extensiveMountPointCheck {
```

Comment:
@jsafrane, what do you think of not even checking if it's a mount point before trying to remove the directory?

If it's still mounted or populated, then os.Remove should fail.

Reply:
Good point, `os.Remove` should fail. So we perhaps don't need mount check at all.